language/compgen/nqg/model/qcfg/qcfg_parser.py

```diff
@@ -124,12 +124,9 @@ def node_fn(unused_span_begin, unused_span_end, rule, children):
 
   def postprocess_cell_fn(nodes):
     """Filter and merge generated nodes."""
-    new_nodes = []
-    for node in nodes:
-      # Discard targets that are not substrings of the gold target.
-      if node in target:
-        new_nodes.append(node)
-    return list(set(new_nodes))
+    new_nodes =  list(set(nodes).intersection(target))
```

Comment:
This changes the behavior of this function.

Consider this example:
target = "foo bar"
nodes = ["foo", "bar", "abc"]

The original code will return a list containing "foo" and "bar", as they are substrings of `target`. The change here will return an empty list, as it checks for exact match.